### PR TITLE
VZ-11521: Build using golang 1.20.10 and OL8 containers

### PR DESCRIPTION
**Pull request by mgianatagh**

### CODE REVIEW

File: package/Dockerfile [hunk 1]
```diff
@@ -1,20 +1,26 @@
...
+    microdnf install -y git && \
+    cd /go/src/github.com/rancher/fleet/scripts && \
+    ./build && \
+    ./test
+
+FROM ghcr.io/oracle/oraclelinux:8-slim
+ARG ARCH
+ENV ARCH=$ARCH
+
+RUN microdnf update -y --setopt=install_weak_deps=0 --setopt=tsflags=nodocs && \
```
Comment (jmaron99) [inline package/Dockerfile:18]: noticed docker-cli and tar are not included explicitly in microdnf calls.  Are they available in the base image or not needed?

Comment (jmaron99) [reply package/Dockerfile:18]: Note that tar is used by 'kubectl cp' so I wonder if it's needed in certain fleet scenarios?

Comment (mgianatagh) [reply package/Dockerfile:18]: they are in the base image

**Review by jmaron99 — APPROVED** [review]

File: package/Dockerfile (hunk 1 above)
Comment (bobdonat) [reply package/Dockerfile:18]: same with curl and ca-certificates

**Pull request merged**